### PR TITLE
feat: add `priority` key

### PR DESCRIPTION
**Pull request by Elsie19**
## Purpose

It allows more control for package maintainers.

## Addendum

~~@oklopfer wants this because it would prevent many cases of unicorn deleting itself.~~
It can be used for who knows what but it's gonna be added so yeah.

## Checklist

- [x] I confirm that I have read the [contributing guidelines](https://github.com/pacstall/pacstall/blob/develop/CONTRIBUTING.md), and this pull request is abiding by all the clauses stated in the guideline.

### CODE REVIEW

File: misc/scripts/install-local.sh [hunk 1]
```diff
@@ -785,6 +785,15 @@ if ! checks; then
     return 1
 fi
 
+# If priority exists and is required, and also that this package has not been installed before (first time)
+if [[ -n ${priority} && ${priority} == 'required' ]] && ! is_package_installed "${name}"; then
```
Comment (D-Brox) [inline misc/scripts/install-local.sh:789]: Wouldn't this be enough?
```suggestion
if [[ ${priority} == 'required' ]] && ! is_package_installed "${name}"; then
```

Comment (Elsie19) [reply misc/scripts/install-local.sh:789]: No. We need to check that `priority` exists first, not assume it does.

Comment (D-Brox) [reply misc/scripts/install-local.sh:789]: If it doesn't exist, then it is interpreted as blank, which is different from `"required"`

Comment (Elsie19) [reply misc/scripts/install-local.sh:789]: I'd like it to still be more explicit.

**Comment by oklopfer** [comment]
how does this actually solve our rhino dependency problems tho, if the `-core` packages are meant to replace each other? would it not fail out when trying to swap them?

**Comment by D-Brox** [comment]
@oklopfer IIRC the replaces field takes care of that. Needs testing tho

**Comment by Elsie19** [comment]
Note: the key we need is `Essential` for some reason. We can have both though, it's no big deal.

File: misc/scripts/install-local.sh [hunk 2]
```diff
@@ -785,6 +785,15 @@ if ! checks; then
...
+    ask "This package has 'priority=required', meaning once this is installed, it should be assumed to be uninstallable. Do you want to continue?" Y
+    if ((answer != 0)); then
+        cleanup
+        exit 1
```
Comment (oklopfer) [inline misc/scripts/install-local.sh:793]: this prompt doesnt work btw it just kills the entire script regardless of response

**Review by oklopfer — CHANGES_REQUESTED** [review]

File: misc/scripts/install-local.sh (hunk 2 above)
Comment (Elsie19) [reply misc/scripts/install-local.sh:793]: Huh? How?

**Comment by oklopfer** [comment]
https://github.com/Elsie19/pacstall/pull/5

**Comment by Elsie19** [comment]
Continued from https://github.com/Elsie19/pacstall/pull/5

**Review by oklopfer — APPROVED** [review]
tested LGTM 🚢

**Pull request merged**